packages/abi-coder/src/coders/array.ts

```diff
@@ -34,6 +35,10 @@ export class ArrayCoder<TCoder extends Coder> extends Coder<
   }
 
   decode(data: Uint8Array, offset: number): [DecodedValueOf<TCoder>, number] {
+    if (data.length < this.encodedLength || data.length > MAX_BYTES) {
+      this.throwError(ErrorCode.DECODE_ERROR, `Invalid array data size.`);
```

Comment:
I don't see much merit in the existence of `this.throwError(...)` because internally it's just a wrapper around `new FuelError`. It's one one-liner replacing another one-liner.

However, what could be done to make this more meaningful is to change `throwError`:
```ts
  throwError(type: "encode" | "decode", message: string): never {
    throw new FuelError(type === "encode" ? ErrorCode.ENCODE_ERROR : DECODE_ERROR, message);
  }
```

This suggestion is only to make the `throwError` method have a reason to exist. I'd also be fine with removing it totally and replacing its invocations with `new FuelError(...)`.

Reply:
I think my preference is to get rid of the function, it does seem unnecessary.